document/web-api/index.bs

```diff
@@ -148,15 +153,49 @@ The [=serialization steps=], given |value|, |serialized|, and |forStorage|, are:
 
     1. Set |serialized|.\[[AgentCluster]] to the [=current Realm=]'s corresponding [=agent cluster=].
 
+    1. Let |rooted| be false.
+
+    1. If |value|.\[[ModuleRecord]] is not null and |value|.\[[ModuleRecord]].\[[HostDefined]] is not empty,
+
+        1. Let |moduleScript| be |value|.\[[ModuleRecord]].\[[HostDefined]].
+
+        1. Set |rooted| to |moduleScript|'s [=rooted source=] boolean.
+
+        1. If |rooted| is true then,
+
+            1. Set |serialized|.\[[Rooted]] to true.
+
+            1. Set |serialized|.\[[BaseURL]] to |moduleScript|'s [=base URL=].
+
+    1. If |rooted| is false,
+
+        1. Set |serialized|.\[[Rooted]] to false.
+
+        1. Set |serialized|.\[[BaseURL]] to null.
+
 The [=deserialization steps=], given |serialized|, |value|, and |targetRealm| are:
 
+    1. If |targetRealm|'s corresponding [=agent cluster=] is not |serialized|.\[[AgentCluster]], then throw a "<a exception>DataCloneError</a>" {{DOMException}}.
+
     1. Let |bytes| be the [=sub-deserialization=] of |serialized|.\[[Bytes]].
 
     1. Set |value|.\[[Bytes]] to |bytes|.
+    
+    1. [=Compile a WebAssembly module=] from |bytes| and set |value|.\[[Module]] to the result.
 
-    1. If |targetRealm|'s corresponding [=agent cluster=] is not |serialized|.\[[AgentCluster]], then throw a "<a exception>DataCloneError</a>" {{DOMException}}.
+    1. Assert: there was no compile error as this same module was previously compiled successfully.
 
-    1. [=Compile a WebAssembly module=] from |bytes| and set |value|.\[[Module]] to the result.
+    1. Let |rooted| be the [=sub-deserialization=] of |serialized|.\[[Rooted]].
+
+    1. If |rooted| is true then,
+
+        1. Let |baseURL| be the [=sub-deserialization=] of |serialized|.\[[BaseURL]].
+
+        1. Let |settings| be the [=current settings object=].
+
+        1. Let |fetchOptions| be the [=default script fetch options=].
+
+        1. Let <var ignore>moduleScript</var> be the result of [=creating a WebAssembly module script=] given |value|.\[[Module]], |settings|, |baseURL|, |fetchOptions|, true.
```

Comment:
Why assign this to a variable that is ignored anyway?

Also, it should probably be "and true" at the end.

Reply:
Thanks for the suggestions. The alternative here would be I think just:

```suggestion
        1. [=Create a WebAssembly module script=] given |value|.\[[Module]], |settings|, |baseURL|, |fetchOptions|, and true.
```

where this operation will then in turn set the `[[ModuleRecord]]` slot on the module object. How does that sound?

Reply:
Yeah that makes more sense. The way it's worded currently makes it seem like something is missing.